test/Microsoft.Azure.WebJobs.Host.UnitTests/Executors/TriggeredFunctionExecutorTests.cs

```diff
@@ -74,15 +76,76 @@ public async Task TryExecuteAsync_WithInvokeHandler_InvokesHandler()
             Assert.True(innerInvokerInvoked);
         }
 
+        [Theory]
+        [InlineData(true, null)]
+        [InlineData(true, "TestSharedListener")]
+        [InlineData(false, null)]
+        public async Task TryExecuteAsync_DynamicConcurrencyEnabled(bool dynamicConcurrencyEnabled, string sharedListenerFunctionId)
+        {
+            var triggerData = new TriggeredFunctionData
+            {
+                TriggerValue = 123,
+                TriggerDetails = new Dictionary<string, string>()
+            };
+            var functionInstance = (IFunctionInstance)CreateFunctionInstance(Guid.NewGuid(), triggerData.TriggerDetails, false);
+            var functionDescriptor = new FunctionDescriptor();
+            var instaceFactoryMock = new Mock<ITriggeredFunctionInstanceFactory<int>>();
+            instaceFactoryMock.Setup(m => m.Create(It.IsAny<FunctionInstanceFactoryContext<int>>())).Returns(functionInstance);
+
+            var concurrencyOptions = new ConcurrencyOptions
+            {
+                DynamicConcurrencyEnabled = dynamicConcurrencyEnabled
+            };
+            var functionExecutor = GetTestFunctionExecutor(concurrencyOptions: concurrencyOptions);
+            var triggerExecutor = new TriggeredFunctionExecutor<int>(functionDescriptor, functionExecutor, instaceFactoryMock.Object, NullLoggerFactory.Instance);
+
+            string functionId = functionInstance.FunctionDescriptor.Id;
+            
+            if (sharedListenerFunctionId != null)
+            {
+                triggerData.TriggerDetails.Add(LogConstants.SharedListenerFunctionIdKey, sharedListenerFunctionId);
```

Comment:
For extensions like BlobTrigger/Durable they'll flow their listener Function ID along with the rest of the trigger details. I'll be updating the Track 2 BlobTrigger executor to do this.

Reply:
Durable triggers don't currently use `TriggerDetails`, so I guess we'll need to add that. The key in this dictionary always needs to be `LogConstants.SharedListenerFunctionIdKey`, right?  Seems a little weird to have this constant defined in a class called `LogConstants`...

Reply:
Yes, you might decide you want to add additional details in there as well in the future. Other triggers add useful information which flows to AppInsights (e.g. QueueTrigger [here](https://github.com/Azure/azure-webjobs-sdk/blob/3f4ec78be9f43bb041937425ced00b341883aa42/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Listeners/QueueTriggerExecutor.cs#L39)). Regarding this constants class - I put it there because we're already using this for other data in TriggerDetails, e.g. [here](https://github.com/Azure/azure-webjobs-sdk/blob/524decf320ca9cf8a35f10b675af7b4a36ecc1c5/src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/Constants/LogConstants.cs#L178). @brettsam - is this an OK place for this key?

Reply:
Changed the model. Due to other issues I discovered, I've taken another approach. The shared ID is no longer flowed via trigger details, it's specified declaratively via a new SharedListenerAttribute applied to the ITriggerBinding implementation for the extension.